shipment_advice/models/shipment_advice.py

```diff
@@ -295,64 +304,96 @@ def _lock_records(self, records):
         self.env.cr.execute(sql, (tuple(records.ids),), log_exceptions=False)
 
     def action_done(self):
-        wiz_model = self.env["stock.backorder.confirmation"]
+        self._check_action_done_allowed()
         for shipment in self:
-            if shipment.state != "in_progress":
+            shipment._action_done()
+        return True
+
+    def _action_done(self):
+        # Validate transfers (create backorders for unprocessed lines)
+        self.ensure_one()
+        self.write({"state": "in_process", "error_message": False})
+
+        if self.shipment_type == "incoming":
+            pickings = self.planned_picking_ids
+            backorder_policy = "create_backorder"
+        else:
+            pickings = self.loaded_picking_ids
+            backorder_policy = self.company_id.shipment_advice_outgoing_backorder_policy
+        pickings = pickings.filtered(lambda p: p.state not in ("cancel", "done"))
+        if self.run_in_queue_job:
+            chain(
+                group(
+                    *[
+                        self.delayable()._validate_picking(picking, backorder_policy)
```

Comment:
Can you add `identity_key=identity_exact` to avoid duplicated jobs?

Reply:
done